physical/raft/transaction.go

```diff
@@ -715,6 +720,7 @@ func (t *RaftTransaction) Commit(ctx context.Context) error {
 	// the transaction application in Raft, applyLog will gather it for us
 	// and return it as a proper error.
 	t.b.l.RLock()
+	log.LowestActiveIndex = pointerutil.Ptr(min(lowestActiveIndex, t.b.raft.AppliedIndex()-1)) // we need to cap the lowest active index, otherwise we might miss transaction started later
```

Comment:
```suggestion
	appliedIndex := t.b.raft.AppliedIndex()
	// Prevent underflow: only set if AppliedIndex > 0
	if appliedIndex > 0 {
		log.LowestActiveIndex = pointerutil.Ptr(min(lowestActiveIndex, appliedIndex-1)) // we need to cap the lowest active index, otherwise we might miss transaction started later
	}
```

Should we check here at first if appliedIndex > 0 ?

Reply:
I don't think so. 0 is a valid index in raft (read: in the implementation of raft we are using)

Reply:
But does this mean, if appliedIndex == 0, we will clean up everything, as we will get the LowestActiveIndex max value of uint64?

Reply:
@phil9909 I've checked https://go.dev/play/p/a0dASDoA3Jn - okay in this case we will get back lowestActiveIndex.